client/src/AddVideoForm.jsx

```diff
@@ -0,0 +1,98 @@
+
+import React, { useState } from 'react';
+import { FormProvider, useForm } from 'react-hook-form'
+
+const AddVideoForm = ({video, onAddVideo }) => {
+  const [title, setTitle] = useState('');
+  const [url, setUrl] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const methods = useForm()
+  // const [success, setSuccess] = useState(false)
+
+    const handleCancel = () => {
+    setTitle('');
+    setUrl('');
+  };
+
+    const validateUrl = (url) => {
+    // Simple validation for YouTube URLs
+    return /^https?:\/\/(www\.)?youtube\.com\/watch\?v=[\w-]+$/.test(url);
+  };
+
+  const handleAddVideo = () => {
+
+    if (title.trim() === '') {
+      setErrorMessage('Please enter a title');
+      return;
+    }
+
+    if (!validateUrl(url)) {
+      setErrorMessage('Please enter a valid YouTube URL');
+      return;
+    }
+
+    const newVideo = {
+      id: Date.now(), // Generate a unique ID
+      title: title,
+      url: url,
+      rating: video.rating,
+    };
+    onAddVideo(newVideo);
+    setTitle('');
```

Comment:
This looks very similar to the code in `handleCancel` - could you maybe rename that function to reflect what it _does_ rather than when you call it, and also call it here?